crates/mcp-apollo-server/src/explorer.rs

```diff
@@ -43,17 +47,42 @@ impl Explorer {
     }
 
     fn create_explorer_url(&self, input: &Value) -> String {
-        let compressed = lz_str::compress_to_uint8_array(input.to_string().as_str());
-        let encoded = base64::engine::general_purpose::STANDARD.encode(compressed);
+        let mut input = input.clone();
+
+        let document = input.get("document").and_then(|v| v.as_str());
+        if document.is_none() || document == Some("") {
+            if let Some(obj) = input.as_object_mut() {
+                obj.insert("document".to_string(), Value::String("{}".to_string()));
+            }
+        }
+        let variables = input.get("variables").and_then(|v| v.as_str());
+        if variables.is_none() || variables == Some("") {
+            if let Some(obj) = input.as_object_mut() {
+                obj.insert("variables".to_string(), Value::String("{}".to_string()));
+            }
+        }
+        let headers = input.get("headers").and_then(|v| v.as_str());
+        if headers.is_none() || headers == Some("") {
+            if let Some(obj) = input.as_object_mut() {
+                obj.insert("headers".to_string(), Value::String("{}".to_string()));
+            }
+        }
```

Comment:
Would it make sense to just use serialize / deserialize the `Input` struct with default values if empty? The struct is already defined above and using it would allow us to remove the `#[allow(dead)]`.